### PR TITLE
Replace windows io include for C standard library

### PR DESCRIPTION
**Pull request by galanCA**
## Description

Replace all io includes for standard c libraries functions. The file following file will be modified:
- [x] brfw.cpp: Replace the use of write, lseek, open, close and add a FILE pointer.
  - BinFile::create()
  - BinFile::close()
  - BinFile::write()
  - Note the class BinFile is only used for the ResfWriter class which is not at all used in CSE. There is also a note that it will crash the program. The PR author proposes to remove ResfWriter and BinFile.
- [x] impf.cpp: Replace the use of write, lseek, tell, open, close and add a FILE pointer.
  - Replace type declaration for fh in CNRECS.DEF. 
  - IMPF::close(): Replace windows close for `fclose`. Also all  values affected in this functions 
  - impfStart(): Replace the functions 
    - ::open(RDONLY|BINARY) for fopen("rb")
    - tell for ftell()
  - impfAfterWarmup: sleek for fseek and change the error check of newPos for non-zero value
  - IMPF::readBuf(): read replace for fread(). this change also requires to change nRead for size_t to prevent loss of precision. TODO ask: Do I need to multiply the result by sizeof(char)?
- [x] yacam.h: 
  - Change mFh from int to FILE*.
  - Change return type for method fh from int to FILE*.
  - if out create and write.
- [x] yacam.cpp: YACAM::
  - open(): Change open function for fopen(). The options are either binary read and write or binary read.
  - close(): Change close for fclose also modify mFh type as needed.
  - rewind(): Chance fseek and as well as the return value check. The inputs are the same.
  - clrBufIf(): Change fseek as well as their return check. change fwrite and its inputs.
  - Replace the use of open, close, lssek, write, read and file id.
- [x] wcmp.cpp: These changes are not currently in use since cmakelist gives three inputs instead of 2.
  - Change the input types for function filesAreDiffBinary to FILE*.
  - Replace a windows function to check the length of the file for seeking the end of a file and returning the number of bytes.
  - Replace window io functions _open_s, _read, _close, for their respective C standard libraries.

## Test

The following test has been created to ensure it maintains its previous functionality.

- [x] brfw.cpp: No test needed as it will be if-out of CSE and not used at all. Ran a local ctest by deleting brfw.h, no changes were found.
- [x] brfw.h: Tested by deleting brfw.h and running local ctest.
- [x] impf.cpp:  All the changes made are active when `IMPORTFILE` is used such as `RLUP( ImpfB, impf) // loop import files` function loops over the `IMPORTFILE`(s). The test used to check the changes was pv_bt_test, which contains IMPORTFILE.
  - pv_bt_test is able to access fopen(); fseek(); ftell; fclose(). And they behave as their predecessors.
  - A test failed_import_file was created to test not containing the files. 
    - Error message created: `importFile 'Data': S0575: Unused IMPORTFILE: no IMPORT()s from it found.`
- [x] yacam.cpp: The following test were tested by stepping through using the debuger, also they pass ctest.
    - ::init
    - ::open
    - ::clrBufIf
    - ::close
  - The following tests are not stopped during the run test. Thus they are if-out from CSE, until future use:
    - ::write
    - ::create
- [x] wcmp.cpp: 
  - To test the functions the author step through the changes and test with different and similar files. The output was as expected.
  - Note regression test on ctest has no effect on these changes. 

## Author Progress Checklist:

- [ ] Open draft pull request
    - [ ] Make title clearly understandable in a standalone change log context
    - [ ] Assign yourself the issue
    - [ ] Add at least one label (enhancement, bug, or maintenance)
    - [ ] Link the issue(s) addressed by this PR (under "Development" in the sidebar menu) 
- [ ] Make code changes (if you haven't already)
- [ ] Self-review of code
    - [ ] My code follows the style guidelines of this project
    - [ ] I have added comments to my code, particularly in hard-to-understand areas
    - [ ] I have only committed the necessary changes for this fix or feature
    - [ ] I have made corresponding changes to the documentation
    - [ ] My changes generate no new warnings
    - [ ] I have ensured that my fix is effective or that my feature works as intended by:
        - [ ] exercising the code changes in the test framework, and/or
        - [ ] manually verifying the changes (as explained in the the pull request description above)
    - [ ] My changes pass all local tests
    - [ ] My changes successfully passes CI checks
    - [ ] Add any unit test for proof and documentation.
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
- [ ] Move pull request out of draft mode and assign reviewers
- [ ] Iterate with reviewers until all changes are approved
    - [ ] Make changes in response to reviewer comments
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
    - [ ] Re-request review in GitHub

## Reviewer Checklist:

 - [ ] Read the pull request description
 - [ ] Perform a code review on GitHub
 - [ ] Confirm all CI checks pass and there are no build warnings
 - [ ] Pull, build, and run automated tests locally
 - [ ] Perform manual tests of the fix or feature locally
 - [ ] Add any review comments, if applicable
 - [ ] Submit review in GitHub as either
     - [ ] Request changes, or
     - [ ] Approve
 - [ ] Iterate with author until all changes are approved

### CODE REVIEW

**Comment by galanCA** [comment]
This PR is ready, however, it depends on PR #289 being merged to add a new unit test.

File: src/brfw.cpp [hunk 1]
```diff
@@ -12,14 +12,13 @@
 
 #include <fcntl.h>
 #include <sys/stat.h>
-#include <io.h>			// _creat
 
 #include "ancrec.h"		// record: base class for rccn.h classes
 #include <rccn.h>		// MTR, MTR_IVL_SUB
 #include "msghans.h"		// MH_R1960
-
+#ifdef BINRES //cnglob.h
```
Comment (nealkruis) [inline src/brfw.cpp:19]: You don't need these in this file, since it isn't being compiled.

File: src/brfw.cpp [hunk 2]
```diff
@@ -30,6 +29,7 @@ static BOO badSrcOrD( void *src, void *d, const char *fcnName);
 const char * FC pointExt( const char *pNam);
 const char * FC pointNamExt( const char *pNam);
 //---------------------------------------------------------------------------
+#ifdef BINRES // TODO (MP) Windows io header was replaced for C-style file io fopen.fclose, etc. These changes were not tested.
```
Comment (nealkruis) [inline src/brfw.cpp:32]: You don't need these in this file, since it isn't being compiled.

File: src/brfw.h [hunk 3]
```diff
@@ -15,22 +15,23 @@
 const char * FC pointExt( const char *pNam);
 const char * FC pointNamExt( const char *pNam);
 //---------------------------------------------------------------------------
+#ifdef BINRES  // TODO (MP) Windows io header was replaced for C-style file io fopen.fclose, etc. These changes were not tested.
```
Comment (nealkruis) [inline src/brfw.h:18]: You don't need these in this file, since it isn't being included if BINRES is defined.

File: src/cnguts.cpp [hunk 4]
```diff
@@ -34,8 +34,9 @@
 #include <cnewin.h>		// BrHans -- if needed here
 #endif
 #endif
+#ifdef BINRES //cnglob.h
```
Comment (nealkruis) [inline src/cnguts.cpp:37]: This is where you want to say:

```
// TODO (MP) Windows io header was replaced for C-style file io fopen.fclose, etc. These changes were not tested.
```

I assume the `//cnglob.h` is supposed to indicate this is defined (or not) in cnglob.h. But we had also talked about making BINRES a CMake option. Where did we land on that? I think I have a slight preference to make it a CMake option (then we can include the TODO (MP) there).

File: src/wfpak.cpp [hunk 5]
```diff
@@ -1819,11 +1819,13 @@ RC WFILE::wf_Close( 		// Close weather file if open
 {
 	RC rc = RCOK;
 	if (yac)
-	{	if ( hdr				// if header given
+	{
```
Comment (nealkruis) [inline src/wfpak.cpp:1822]: I thought we agreed we wouldn't use SHAs in comments like this. This should be a TODO (MP) like everything else.

File: src/yacam.cpp [hunk 6]
```diff
@@ -1040,6 +967,77 @@ RC YACAM::errFlLn( const char *s, ...)	// error message "%s in <mWhat> <mPathNam
 			   (char *)MH_I0118,		// "%s '%s' (near line %d):\n  %s"
 			   mWhat,  mPathName ? mPathName : "bug",  mLineNo,  buf );
 }
+#if 0 // The methods below are not used but have been modified to fit c standard file io. SHA e6181ff06f42350c8c559211c3809e25a03d89f2
```
Comment (nealkruis) [inline src/yacam.cpp:970]: Same here...should be TODO (MP).

File: src/yacam.h [hunk 7]
```diff
@@ -88,6 +85,11 @@ class YACAM
 
     int mErOp;	// communicates erOp from entry points to error fcns
 				// note need a data mbr at end due to rcdef.exe deficiency 10-94.
+#if 0 // The methods below are not used but have been modified to fit c standard file io. SHA e6181ff06f42350c8c559211c3809e25a03d89f2
```
Comment (nealkruis) [inline src/yacam.h:88]: Same here...TODO (MP).

File: test/wcmp/wcmp.cpp [hunk 8]
```diff
@@ -47,27 +44,41 @@ int main( int argc, char **argv)
 }		// main
 //---------------------------------------------------------------------------
 static int filesAreDiffBinary( 	// binary compare file length and contents
```
Comment (nealkruis) [inline test/wcmp/wcmp.cpp:46]: I don't think we are using this function anywhere in our tests (@chipbarnaby ?), so maybe this should have a TODO (MP) flag too?

**Review by nealkruis — CHANGES_REQUESTED** [review]
Mostly good, but still some wrinkles to iron out.

File: test/wcmp/wcmp.cpp (hunk 8 above)
Comment (galanCA) [reply test/wcmp/wcmp.cpp:48]: The flag is written above the function. Are there other places where you would want the tag? Like the place where the function is used.

File: src/cnguts.cpp [hunk 9]
```diff
@@ -30,20 +30,21 @@
...
 #ifdef WINorDLL
-#ifdef BINRES //cnglob.h
+#ifdef BINRES 		// CMake option
 #include <cnewin.h>		// BrHans -- if needed here
 #endif
 #endif
+#ifdef BINRES 		// CMake option
 #include "brfw.h"	// ResfWriter
-
+#endif
```
Comment (nealkruis) [inline src/cnguts.cpp:39]: This can be cleaner.

File: test/wcmp/wcmp.cpp [hunk 10]
```diff
@@ -47,27 +44,41 @@ int main( int argc, char **argv)
...
+		len1 = ftell(hFile1);
+		len2 = ftell(hFile2);
+		if (fseek(hFile1, 0, SEEK_SET) != 0 || fseek(hFile2, 0, SEEK_SET) != 0) {
+			return 2;
+		}
+	}
+	else {
+		return 2;
+	}
+
```
Comment (nealkruis) [inline test/wcmp/wcmp.cpp:72]: Why not make a "get_file_length" function?

**Review by nealkruis — CHANGES_REQUESTED** [review]

**Review by nealkruis — APPROVED** [review]

**Pull request merged**